From 6c0c9fd6a049e8c2a456e39dd94cb7c2c65203db Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Tue, 4 Aug 2020 11:57:53 +0530
Subject: [PATCH 618/767] octeontx2-pf: Avoid null pointer dereference

commit 92fd56ae53e27c336da9e15c7c5b6e4852d95a0d from
git@git.assembla.com:cavium/WindRiver.linux.git

otx2_mbox_up_handler_cgx_ptp_rx_info can be
called for a VF too and VF do not have ptp
context. Hence check for valid ptp pointer
before dereferencing.

Change-Id: I30424bc1231ea4384868f22f896933fdab702c1d
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/33392
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 619dddc6d338..7dab96d75ff0 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -873,6 +873,9 @@ int otx2_mbox_up_handler_cgx_ptp_rx_info(struct otx2_nic *pf,
 {
 	int i;
 
+	if (!pf->ptp)
+		return 0;
+
 	pf->ptp->ptp_en = msg->ptp_en;
 
 	/* notify VFs about ptp event */
-- 
2.31.1

